client/jetpack-connect/plans-landing.jsx

```diff
@@ -19,22 +21,48 @@ class PlansLanding extends Component {
 	static propTypes = {
 		basePlansPath: PropTypes.string,
 		interval: PropTypes.string,
+		url: PropTypes.string,
+	};
+
+	static defaultProps = {
+		url: '',
 	};
 
 	componentDidMount() {
 		this.props.recordTracksEvent( 'calypso_jpc_plans_landing_view', {
 			jpc_from: 'jetpack',
 		} );
+
+		this.goToPlansIfAlreadyConnected();
+	}
+
+	componentDidUpdate() {
+		this.goToPlansIfAlreadyConnected();
+	}
+
+	goToPlansIfAlreadyConnected() {
+		const { site } = this.props;
+
+		if ( site ) {
+			page.redirect( '/plans/' + site.slug );
+		}
 	}
 
 	storeSelectedPlan = ( cartItem ) => {
+		const { url } = this.props;
+		let redirectUrl = CALYPSO_JETPACK_CONNECT;
+
+		if ( url ) {
+			redirectUrl += '?url=' + url;
```

Comment:
Don't we have a `addQueryParams` function? Do you think that could have any benefit here?